ls1intum/Apollon_standalone
# `Development`: Improve sharing diagrams feature

### Motivation and Context

This PR enhances the currently existing sharing diagram feature of Apollon Standalone. 
Following are the minor changes that this PR introduces:
1. Users are now redirected directly to the collaboration URL, once they share the diagram in collaboration mode
2. Number of toasters to notify users after sharing the diagram is reduced to one.
3. Description is now added in Collaboration Name Sharing modal
4. Small spacing is added between the collaborators list and theme switcher icon
5. If the user is new, he will now be prompted with Collaboration Name Modal with a random name being assigned to him


### Steps for Testing
To test in test server: 
1. Goto https://test1.apollon.ase.in.tum.de
2. Insert some diagrams in the canvas, and click on the Share button from Menu.
3. Observe that you are automatically redirected to sharing mode URL and sharing link is copied to your clipboard

 
### Screenshot:
#### Redirection to Collaboration URL
![screen-capture (10)](https://user-images.githubusercontent.com/14681902/177195250-ed0a6070-0351-42f7-88b2-686ec595de64.gif)

#### Newly added Description in Collaboration Name Modal
![image](https://user-images.githubusercontent.com/14681902/177196423-fab60dd1-292b-4147-8246-9a362b57584c.png)
#### Updated single Toaster
![image](https://user-images.githubusercontent.com/14681902/177196474-94ed08a9-6b00-4edd-9233-e8e7180a8ff5.png)
#### Spacing between Collaborators list and theme switcher icon
![image](https://user-images.githubusercontent.com/14681902/177196711-97be2720-629c-4ec7-8b74-ef04bd4ba820.png)


#### Todo:
- [x] Assign random name to collaborators
